### DISCUSSION TITLE
🌴🗓️ Annual Leave Application

### DISCUSSION THREAD

**Discussion by Gary-H9** [ministryofjustice/analytical-platform#2472] [Firebreak Winter 23]
### Idea Description 
A Rails application which will allow users to easily track their holidays. 📅

### Why Should We Do This
Tracking of this seems to be very varied _across the organisation_. A simple interface which let's you do the following would be be beneficial:
- login
- track holidays - a calendar interface would be ideal

### Definition of Done
v1: 
- [ ] App is hosted in Cloud Platform
- [ ] App is styled using govuk frontend
- [ ] User can login via SSO - ideally AAD
- [ ] User can personally track their leave and only their leave, no approval required or anything. 
- [ ] A user can see their days remaining / days used 

v2 ideas:
- [ ] The users calendar is viewable by their manager
- [ ] An annual leave request has two states: 'requested' and 'approved' 
- [ ] When a day is requested the users manager is notified 
- [ ] A users manager can then approve a days leave and the state changes

v3 ideas:
- [ ] Teams exist 
- [ ] A user belongs to one team 
- [ ] The team calendar is viewable by members of the team which the user is in

### What skills do we need
Experience in:
- Rails Development
- General programming
- Deploying to Cloud Platform
- Creating containers
- A can do attitude

I think this should be done with Rails as I know this.

**Comment by murdo-moj** [comment]
I'll help with this (disclaimer, I don't know ruby but am willing to learn)

**Reply by murad-ali-MoJ** [reply]
I am interested to learn and contribute.

**Comment by michaeljcollinsuk** [comment]
Interested, just as a refresher on Ruby on Rails as much as anything!

**Comment by ymao2** [comment]
mee too,  if need coder for ruby on rails 🤓
